src/core/unit.c

```diff
@@ -1055,12 +1055,22 @@ static void unit_merge_dependencies(
         if (u == other)
                 return;
 
+        /* First, remove dependency to other. */
+        HASHMAP_FOREACH_KEY(deps, dt, u->dependencies) {
+                UnitDependencyInfo di_other;
```

Comment:
`di_other` is initialized, but it isn't really used for anything. Maybe write this as
```C
if (hashmap_remove(deps, other) != NULL)
    unit_maybe_warn_about_dependency(u, other->id, UNIT_DEPENDENCY_FROM_PTR(dt));
```
?